client/human/HumanClientApp.cpp

```diff
@@ -625,16 +627,29 @@ void HumanClientApp::CancelMultiplayerGameFromLobby()
 { m_fsm->process_event(CancelMPGameClicked()); }
 
 void HumanClientApp::SaveGame(const std::string& filename) {
-    m_save_game_in_progress = true;
-    Message response_msg;
+    m_game_saves_in_progress.push(filename);
+
+    // Start a save if there is not one in progress
+    if (m_game_saves_in_progress.size() > 1) {
+        DebugLogger() << "Add pending save to queue.";
+        return;
+    }
+
     m_networking->SendMessage(HostSaveGameInitiateMessage(filename));
-    DebugLogger() << "HumanClientApp::SaveGame sent save initiate message to server...";
+    DebugLogger() << "Sent save initiate message to server.";
 }
 
 void HumanClientApp::SaveGameCompleted() {
-    DebugLogger() << "HumanClientApp::SaveGameCompleted by server.";
-    m_save_game_in_progress = false;
-    SaveGameCompletedSignal();
+    m_game_saves_in_progress.pop();
+
+    // Either indicate that all save are completed or start the next save.
```

Comment:
In the current game if the user saves while the autosave is in progress.

It only matters if the user saves and then immediately exits.  In the current game that means only the first save that completes is kept, the autosave.  The user save is always aborted.  

With PR #1671, the scenario would happen whenever the user saves and then immediately quits.  Although in that case the last turn autosave would be lost.